File: web-api-batch-28a---backend-Ramudgar/models/job.js
const mongoose = require("mongoose");
const user=require('../models/Users');

let schema = new mongoose.Schema(
  {
    userId:{
      type: mongoose.Schema.Types.ObjectId,
      ref: "users",
    },
    
    title: {
      type: String,
      required: true,
    },
    maxApplicants: {
      type: Number,
    },
    maxPositions: {
      type: Number,
    },
    location: {
      type: String,
      required: true,
    },
    
    dateOfPosting: {
      type: String,
      default: Date.now,
    },
    deadline: {
      type: String,
      
    },
    skills: {
      type: String,   
    },
    jobType: {
      defalult: "PartTime",
      enum: ["FullTime", "PartTime", "Contract"],
      type: String,
      required: true,
    },
    duration: {
      type: String, 
    },
    salary: {
      type: String, 
    },
    description:{
      type: String,
    },
    image:{ type: String ,
      },
    status: {
      type: String,
      enum: ["Accepted", "Rejected","Pending"],
      default: "Pending",
    },
    jobApplicants:[{
        type: mongoose.Schema.Types.ObjectId,
        ref: "users",
      }],
      jobApplicantsCount: {
        type: String,
      },
      apply: {
       type:String,  }
  },
  { timestamps: true } 
);
module.exports = mongoose.model("jobs", schema);

File: final-assignment-flutter/lib/model/job.dart
import 'package:json_annotation/json_annotation.dart';

part 'job.g.dart';

@JsonSerializable()
class Job {
  @JsonKey(name: '_id')
  String? id;
  String? title;
  num? maxApplicants;
  num? maxPositions;
  String? location;
  String? dateOfPosting;
  String? deadline;
  String? skills;
  String? duration;
  String? salary;
  num? rating;
  String? jobType;
  // String? category;
  String? image;
  String? description;
  String? status;
  String? apply;

  Job({
    this.id,
    this.title,
    this.maxApplicants,
    this.maxPositions,
    this.location,
    this.dateOfPosting,
    this.deadline,
    this.skills,
    this.duration,
    this.salary,
    this.rating,
    this.jobType,
    // this.category,

    this.image,
    this.description,
    this.status,
    this.apply,
  });

//3. flutter pub run build_runner build
  factory Job.fromJson(Map<String, dynamic> json) {
    return _$JobFromJson(json);
  }

  Map<String, dynamic> toJson() => _$JobToJson(this);
}
